netsgiro/records.py

```diff
@@ -15,40 +15,65 @@
     'AvtaleGiroAmountItem1',
     'AvtaleGiroAmountItem2',
     'AvtaleGiroSpecification',
+    'AvtaleGiroAgreement',
     'get_records',
 ]
 
 
 def to_service_code(
         value: Union[netsgiro.ServiceCode, int, str]) -> netsgiro.ServiceCode:
-    if isinstance(value, netsgiro.ServiceCode):
-        return value
     return netsgiro.ServiceCode(int(value))
 
 
 def to_record_type(
         value: Union[netsgiro.RecordType, int, str]) -> netsgiro.RecordType:
-    if isinstance(value, netsgiro.RecordType):
-        return value
     return netsgiro.RecordType(int(value))
 
 
+def to_assignment_type(
+        value: Union[netsgiro.AvtaleGiroAssignmentType, int, str]
+        ) -> netsgiro.AvtaleGiroAssignmentType:
+    return netsgiro.AvtaleGiroAssignmentType(int(value))
+
+
 def to_avtalegiro_transaction_type(
         value: Union[netsgiro.AvtaleGiroTransactionType, int, str]
         ) -> netsgiro.AvtaleGiroTransactionType:
-    if isinstance(value, netsgiro.AvtaleGiroTransactionType):
-        return value
     return netsgiro.AvtaleGiroTransactionType(int(value))
 
 
-def to_date(value: Union[datetime.date, str]) -> Optional[datetime.date]:
+def to_avtalegiro_registration_type(
+        value: Union[netsgiro.AvtaleGiroRegistrationType, int, str]
+        ) -> netsgiro.AvtaleGiroRegistrationType:
+    return netsgiro.AvtaleGiroRegistrationType(int(value))
+
+
+def to_date(value: Union[datetime.date, str, None]) -> Optional[datetime.date]:
     if isinstance(value, datetime.date):
         return value
-    if value == '000000':
+    if value == '000000' or value is None:
```

Comment:
Reverse the conditions, makes slighty more sense and maximises the performance to 11